scripts/pr.py

```diff
@@ -0,0 +1,273 @@
+
+home = expanduser("~")
+configPath = home + '/.pr_script_config.json'
+scriptPath = os.path.realpath(__file__)
+
+
+def requirePass():
+    return getpass.getpass("Type your Github password: ")
+
+
+def requireBase():
+    return raw_input("Please specify the base branch you want the changes pulled into: ")
+
+
+def requireTickedId():
+    return raw_input("Type JIRA ticked's id: ")
+
+
+def requireTickedHeader():
+    return raw_input("Type JIRA ticked's description: \n")
+
+
+def requireBody():
+    return raw_input("Type PR's body: \n")
+
+
+def requireRemote():
+    remote = raw_input("Please, specify the repository for PR's in format ':owner/:repo': \n")
+    return remote.lower()
+
+
+def requireFeatureName():
+    return raw_input("Type name of the feature you will work on: ").replace(" ", "-")
+
+
+def mapDict(f,d):
+    return dict(map(f, d.iteritems()))
+
+
+def validateValues(items):
+    (k,v) = items
+    if v is None:
+        print("Cannot infer the {0} value".format(k))
+        newV = raw_input("Please specify a value for the : {0}".format(k))
+        return (k,newV)
+    else:  
+        return (k,v)
+
+    
+def readFile(name):
+    f = open(name, 'a+')
+    content = f.read()
+    f.close()
+    return content
+
+        
+def execStdout(cmd):
+    try:
+        return subprocess.check_output(cmd)
+    except subprocess.CalledProcessError:
+        None
+
+
+def mapOptional(x,f):
+    if not x is None:
+        return f(x)
+    else:
+        x
+
+
+def gitName():
+    name = execStdout(["git", "config", "user.name"])
+    mapOptional(name, lambda x: x.stript())
+
+
+def gitCurrentBranch():
+    return execStdout(["git", "rev-parse", "--abbrev-ref", "HEAD"]).strip()
+
+
+def gitOrigin():
+    origin =  execStdout(["git", "config", "remote.origin.url"])
+    mapOptional(origin, lambda x: x.strip())
+
+
+def parseJson(content):
+    try:
+        return json.loads(content)
+    except ValueError:
+        return {}
+
+    
+def optionalGet(dict,k):
+    if dict is None:
+        None
+    else:
+        dict.get(k, None)
+
+
+def dictMerge(l,r):
+    res = l.copy()
+    res.update(r.iteritems())
+    return res
+
+
+def writeJson(dict, name):
+    f = open(name, 'w')
+    content = json.dumps(dict)
+    f.write(content)
+    f.close()
+
+    
+def execCmd(args):
+    return subprocess.call(args)
+                       
+
+    
+class Config:
+    def __init__(self):
+        globalConfig = self.loadConfig()
+        
+        if globalConfig.get(scriptPath, None) is None:
+            newConfig = self.createConfig()
+            self.globalConfig = dictMerge(globalConfig, newConfig)
+            writeJson(self.globalConfig, configPath)
+        else:
+            self.globalConfig = globalConfig
+
+        self.setupConfig()
+
+        
+    def setupConfig(self):
+        self.config = self.globalConfig[scriptPath]
+        for k,v in self.config.items():
+            setattr(self, k, v)
```

Comment:
This might be a little dangerous, since it couples the implementation of this class to the config file itself. Now while writing config file entries, you've got to be sure not to use any of the names of the members of this class, otherwise they'll be overwritten.